L1Trigger/TrackFindingTracklet/test/ProducerIRin.cc

```diff
@@ -0,0 +1,104 @@
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Run.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Utilities/interface/EDPutToken.h"
+#include "FWCore/Utilities/interface/ESGetToken.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DataFormats/Common/interface/Handle.h"
+
+#include "L1Trigger/TrackTrigger/interface/Setup.h"
+
+#include <string>
+#include <vector>
+#include <deque>
+#include <iterator>
+#include <cmath>
+#include <numeric>
+
+using namespace std;
+using namespace edm;
+using namespace tt;
+
+namespace trklet {
+
+  /*! \class  trklet::ProducerIRin
+   *  \brief  Transforms TTTDCinto f/w comparable format for summer chain configuratiotn
```

Comment:
The comment here seems misleading. This class simply extracts the stub streams from the DTC output, and if a reduced L1 track chain is being used, selects just the subset used in that chain and assigns them to specific L1 track board input channels.

Reply:
Throughout this new code, please avoid "summer chain" wherever possible. The summer chain will only exist for another few months, so there's no justification in putting into central CMSSW, summer chain specific code. We should ensure that all such code could handle also a more general "reduced" chain.

Reply:
done.

Reply:
Comment should also mention that this EDProducer can optionally select the subset of the stub streams used in a reduced tracking chain, and connect them to the correct L1 track board input channels.

Reply:
I see done.